android-lib/src/androidTest/java/com/kinvey/androidTest/store/UserStoreTest.java

```diff
@@ -1310,16 +1330,272 @@ private String createRandomUserName(String testName) {
+            @Override
+            public void run() {
+                UserStore.changePassword(password, client, callback);
+            }
+        });
+        looperThread.start();
+        latch.await();
+        looperThread.mHandler.sendMessage(new Message());
+        return callback;
+    }
+
+//  MLIBZ-2653  UserStore#get doesn't return User object
+/*    @Test
+    public void testGet() throws InterruptedException {
+        User user = login(USERNAME, PASSWORD).result;
+        DefaultKinveyClientCallback callback = get(user.getId(), client);
+        assertNull(callback.error);
+        assertNotNull(callback.result);
+        assertEquals(user.getId(), callback.result.getId());
+    }*/
+
+    @Test
+    public void testGetRequestError() throws InterruptedException {
+        login(USERNAME, PASSWORD);
+        DefaultKinveyClientCallback callback = get("unexistUser", client);
+        assertNotNull(callback.error);
+        assertNull(callback.result);
+        assertEquals("This user does not exist for this app backend.",
+                ((KinveyJsonResponseException) callback.error).getDetails().getDescription());
+    }
+
+    private DefaultKinveyClientCallback get(final String userId, final Client client) throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final DefaultKinveyClientCallback callback = new DefaultKinveyClientCallback(latch);
+        LooperThread looperThread = new LooperThread(new Runnable() {
+            @Override
+            public void run() {
+                UserStore.get(userId, client, callback);
+            }
+        });
+        looperThread.start();
+        latch.await();
+        looperThread.mHandler.sendMessage(new Message());
+        return callback;
+    }
+
+    @Test
+    public void testConvenience() throws InterruptedException {
+        User user = login(USERNAME, PASSWORD).result;
+        DefaultKinveyClientCallback callback = convenience(client);
+        assertNull(callback.error);
+        assertNotNull(callback.result);
+        assertEquals(user.getId(), callback.result.getId());
+    }
+
+    @Test
+    public void testConvenienceError() throws InterruptedException {
+        assertFalse(client.isUserLoggedIn());
+        DefaultKinveyClientCallback callback = convenience(client);
+        assertNotNull(callback.error);
+        assertNull(callback.result);
+        assertEquals("currentUser must not be null", callback.error.getMessage());
+    }
+
+    private DefaultKinveyClientCallback convenience(final Client client) throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final DefaultKinveyClientCallback callback = new DefaultKinveyClientCallback(latch);
+        LooperThread looperThread = new LooperThread(new Runnable() {
+            @Override
+            public void run() {
+                UserStore.convenience(client, callback);
```

Comment:
What does the `convenience` function on the `UserStore` for?

Reply:
The convenience function is the similar to `retrieve`, `convenience` gets user metadata from the backend and update `client.activeUser`, `retrieve` doesn't update `client.activeUser`.

Reply:
So in the case of `retrieve`, it only updates the credential store?

Reply:
`retrieve` only returns user data in the callback, but doesn't update the active user.